source/Hardpoint.cpp

```diff
@@ -357,6 +409,18 @@ bool Hardpoint::FireSpecialSystem(Ship &ship, const Body &body, std::vector<Visu
 	if(offset.Length() > range)
 		return false;
 
+	// Check if the target is within the arc of fire.
+	Angle aim(offset);
+	if(!IsOmnidirectional())
+	{
+		Angle minArc = GetMinArc();
+		Angle maxArc = GetMaxArc();
+		minArc += facing;
```

Comment:
Looks like you lost the variable when moving this code in https://github.com/endless-sky/endless-sky/pull/7094/commits/d1114173ed88d2c88da492be587653618f75539e. Just add `Angle facing = ship.Facing();` at the beginning of this function (and replace other uses of ship.Facing() with it).